tests/integration/models/application.spec.js

```diff
@@ -340,6 +340,38 @@ describe('Application Model', function () {
 	});
 
 	describe('given a single application', function () {
+		describe('balena.models.application.rename()', function () {
```

Comment:
May I ask to move this inside the `[read operations]` of the `[contained scenario]` a bit lower, so that we don't get the slowdown from the `givenAnApplication(beforeEach);`?
You can then drop the `givenAnApplication(beforeEach);` and just reuse the app that's already in there and give a different app name for the `['id', 'app_name', 'slug'].forEach` (just add the prop as a suffix to the app name) and then add an `after()` to rename the app back to it's original name.